app/ui/lib/Modal.tsx

```diff
@@ -151,6 +157,8 @@ Modal.Footer = ({
         {cancelText || 'Cancel'}
       </Button>
       <Button
+        type={formId ? 'submit' : 'button'}
+        form={formId}
```

Comment:
This is a hack version, but it would be ideal if the button had no `onClick` and we relied on it triggering the form's `onSubmit`. Unfortunately that would require even an more confusing mix of props.